profiler/src/main/java/com/splunk/opentelemetry/profiler/context/SpanContextualizer.java

```diff
@@ -56,7 +59,10 @@ public void updateContext(RecordedEvent event) {
     if (traceId == null || spanId == null) {
       threadSpans.remove(javaThreadId);
     } else {
-      SpanLinkage linkage = new SpanLinkage(traceId, spanId, javaThreadId);
+      TraceFlags traceFlags = TraceFlags.fromByte(event.getByte("traceFlags"));
```

Comment:
you don't need to pass trace flags with event as it can only be `TraceFlags.getSampled()`

Reply:
Isn't it safer to pass the byte? Users of this distro can configure a custom sampler

Reply:
I'm not sure I follow what's being asked. The JFR event has a byte from the original span. The `SpanContext` requires a `TraceFlags` instance. If the user has a custom sampler, we still want to respect the original value and propagate it accordingly, right?

Reply:
I think the current version is perfectly fine - we should pass that byte, cause it's possible for our users to configure a custom sampler that will sample out some spans.